src/interpreter.c

```diff
@@ -696,7 +696,7 @@ static val_t i_eval_expr(struct v7 *v7, struct ast *a, ast_off_t *pos,
       }
     case AST_INSTANCEOF:
       v1 = i_eval_expr(v7, a, pos, scope);
-      v2 = i_eval_expr(v7, a, pos, scope);
+      v2 = i_value_of(v7, i_eval_expr(v7, a, pos, scope));
```

Comment:
Ah, I see. This will return the cfunction embedded in the callable wrapper. But this cfunction doesn't pass `v7_is_function()`.

Moreover, below the attribute `prototype` is read from `v2` which should be the wrapper, not the cfunc which cannot hold properties.

Why not:

```js
      v1 = i_eval_expr(v7, a, pos, scope);
      v2 = i_eval_expr(v7, a, pos, scope);
      if (!v7_is_function(v2) && !v7_is_cfunction(i_value_of(v7, v2))) {
        throw_exception(v7, "TypeError",
                        "Expecting a function in instanceof check");
      }
      return v7_create_boolean(is_prototype_of(v1,
                                               v7_get(v7, v2, "prototype", 9)));
```